backend/src/github-util.ts

```diff
@@ -151,17 +183,35 @@ const createValidationResult = async (validation_function): Promise<ValidationRe
   return { status: 'valid' };
 };
 
+/** Parse information from portfolio and submission. Returns tuple of earliest
+ *  valid time, deadline, and Idol member github username.
+ *  Raises an error if the Idol member from `submission` does not have a github username. */
+const parsePortfolioSubmission = (portfolio: DevPortfolio, submission: DevPortfolioSubmission) => {
+  const start = portfolio.earliestValidDate;
+  const end = portfolio.deadline;
+  const username = submission.member.github; // must be github username
```

Comment:
the `github` field will be a github URL so we'll probably need to parse the username from the URL